db/pebble/db_metrics.go

```diff
@@ -0,0 +1,110 @@
+package pebble
+
+import (
+	"time"
+
+	"github.com/NethermindEth/juno/db"
+	"github.com/cockroachdb/pebble"
+)
+
+// onCompactionBegin is a callback for the start of a compaction process.
+//
+//nolint:gocritic
+func (d *DB) onCompactionBegin(info pebble.CompactionInfo) {
+	d.compStartTime = time.Now()
+	d.activeComp++
+	if info.Input[0].Level == 0 {
+		d.level0Comp.Add(1)
+	} else {
+		d.nonLevel0Comp.Add(1)
+	}
+}
+
+// onCompactionEnd is a callback for the end of a compaction process.
+//
+//nolint:gocritic
+func (d *DB) onCompactionEnd(info pebble.CompactionInfo) {
+	d.compTime.Add(int64(time.Since(d.compStartTime)))
+	d.activeComp--
+}
+
+// onWriteStallBegin is a callback for the start of a write stall.\
+func (d *DB) onWriteStallBegin(b pebble.WriteStallBeginInfo) {
+	d.writeDelayStartTime = time.Now()
+	d.writeDelayCount.Add(1)
+}
+
+// onWriteStallEnd is a callback for the end of a write stall.
+func (d *DB) onWriteStallEnd() {
+	d.writeDelayTime.Add(int64(time.Since(d.writeDelayStartTime)))
+}
+
+// meter continuously gathers metrics at the specified interval and reports them to the underlying listener.
+func (d *DB) meter(interval time.Duration) {
+	timer := time.NewTimer(interval)
+	defer timer.Stop()
+
+	// Create storage for delta calculation.
+	var (
+		compTimes        [2]time.Duration
+		writeDelayCounts [2]uint64
+		writeDelayTimes  [2]time.Duration
+		compWrites       [2]uint64
+		compReads        [2]uint64
+		nWrites          [2]uint64
+	)
```

Comment:
I don't think you need to do this here, no? Delta calculation can be done on Grafana.

Reply:
I am not sure. The data provided by `Pebble` is the total cumulative so far, and that sounds to me like a case for `Counter` with delta instead of only ever-increasing `gauge`. For example, `compWrites` and `compReads` represent bytes written and read during compactions. I haven't personally seen a bytes-written metric represented as a `gauge` or, even worse, a `counter` that adds the total instead of the change. `go-ethereum` also reports the delta for that, but they use a slightly different collector called meter, which, from what I understand, accumulates data and resets itself after taking a snapshot.

I've been thinking about whether `memComps`, `lvl0Comps`, `nonLvl0Comps`, and `seekComps` shouldn't be represented as counters with a delta, too, for that reason. That would leave us with only three gauges in total: `DiskSize`, `LevelFiles`, and `ManualMemAlloc`. They represent the current value and can decrease or increase between snapshots.

Reply:
> The data provided by Pebble is the total cumulative so far

That is entirely fine, prometheus counters are expected to monotonically increase. They shouldn't be decremented. Let's just export whatever pebble is reporting, I don't think we need to do delta calculation ourselves.

Reply:
Sure, I exported them as they are. Would mind sharing some insights as to why the total? I am also not sure if I shouldn't change the counters into gauges with that change.